Source/WebCore/page/Navigation.cpp

```diff
@@ -40,6 +40,28 @@ Navigation::Navigation(ScriptExecutionContext* context, LocalDOMWindow& window)
 {
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-cangoback
+bool Navigation::canGoBack() const
+{
+    if (hasEntriesAndEventsDisabled())
+        return false;
+    ASSERT(m_currentEntryIndex.has_value());
+    if (!*m_currentEntryIndex)
+        return false;
+    return true;
+}
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-cangoforward
+bool Navigation::canGoForward() const
+{
+    if (hasEntriesAndEventsDisabled())
+        return false;
+    ASSERT(m_currentEntryIndex.has_value());
+    if (*m_currentEntryIndex == m_entries.size() - 1)
```

Comment:
This doesn't match the specification. Seems like it might be a bug in the specification, did you report it?

Reply:
I mentioned this in WHATWG Chat and Domenic will fix it.

Reply:
Thanks, I did not report, but did notice it. I first thought this implied 1 based array indexing rather than zero based, but I checked the spec and indeed it was wrong since lists are specced to be zero based. Anyway nice to see it being fixed!